tools/build-tool/code/tooldebian.pas

```diff
@@ -0,0 +1,192 @@
+unit ToolDebian;
+
+{$I castleconf.inc}
+
+interface
+
+uses
+  CastleFindFiles,
+  ToolArchitectures, ToolManifest, ToolPackage;
+
+type
+  { Package a project to a directory. }
+  TPackageDebian = class(TPackageDirectoryAbstract)
+  strict private
+    { Total size of all binaries in the project, required for Debian metadata }
+    BinariesSize: Int64;
+    { Helper to find total size of all files inside the folder,
+      To satisfy FindFile syntax has to be of Object }
+    procedure FoundFile(const FileInfo: TFileInfo; var StopSearch: Boolean);
+  public
+    procedure Make(const OutputProjectPath: String; const PackageFileName: String;
+      const Cpu: TCpu; const Manifest: TCastleManifest);
+  end;
+
+implementation
+uses
+  SysUtils, Process, {$ifdef UNIX} BaseUnix, {$endif}
+  CastleUtils, CastleFilesUtils, CastleDownload, CastleImages,
+  ToolCommonUtils;
+
+procedure TPackageDebian.FoundFile(const FileInfo: TFileInfo; var StopSearch: Boolean);
+begin
+  BinariesSize += FileInfo.Size;
+end;
+
+procedure TPackageDebian.Make(const OutputProjectPath: String; const PackageFileName: String;
+      const Cpu: TCpu; const Manifest: TCastleManifest);
+
+  function CpuToArchitectureString(const Cpu: TCpu): String;
+  begin
+    case Cpu of
+      { Architectures supported by https://packages.debian.org/buster/bash }
+      x86_64: Result := 'amd64';
+      aarch64: Result := 'arm64';
+      arm: begin
+          Result := 'armel';
+          WriteLn('WARNING: Architecture ' + CpuToString(Cpu) + ' is ambiguous between "armel" and "armhf" in Debian.');
+        end;
+      i386: Result := 'i386';
+      mips: Result := 'mips';
+      mipsel: Result := 'mipsel';
+      powerpc64: begin
+          Result := 'ppc64el';
+          WriteLn('WARNING: Architecture ' + CpuToString(Cpu) + ' is ambiguous between "ppc64el" and "ppc64" in Debian.');
+        end;
+      // "mips64el"
+      // "s390x"
+      { Unofficial Debian ports https://packages.debian.org/sid/bash }
+      m68k: Result := 'm68k';
+      riscv64: Result := 'riscv64';
+      sparc64: Result := 'sparc64';
+      // "alpha"
+      // "sh4"
+      // "x32" = X32 is an ABI for amd64/x86_64 CPUs using 32-bit integers, longs and pointers
+      else {powerpc,sparc,avr,jvm,i8086,riscv32,armeb}
+      begin
+        Result := CpuToString(Cpu);
+        WriteLn('WARNING: Architecture ' + CpuToString(Cpu) + ' is not officially supported by Debian.');
```

Comment:
Use `WriteLnWarning` to report warnings in the build tool.

CGE build tool main program does `ApplicationProperties.OnWarning.Add(@ApplicationProperties.WriteWarningOnConsole)` to make sure that warnings are displayed on console too.

Reply:
Done in 89fb8d670ddeef0fa8863b33811140d9084d2f8c